lib/generator.js

```diff
@@ -32,22 +34,32 @@
     categoriesMeta = {
       categories: categories.sort(comparer).map(extractName)
     };
-    return hexo.route.set('categories.json', JSON.stringify(categoriesMeta));
+
+    return {
+        path: 'categories.json',
+        data: JSON.stringify(categoriesMeta)
+    };
   };
 
   generateTagsJson = function(tags) {
     var tagsMeta;
     tagsMeta = {
       tags: tags.sort(comparer).map(extractName)
     };
-    return hexo.route.set('tags.json', JSON.stringify(tagsMeta));
-  };
 
-  module.exports = function(locals, render, callback) {
-    generatePostsJson(locals.posts.toArray());
-    generateCategoriesJson(locals.categories.toArray());
-    generateTagsJson(locals.tags.toArray());
-    return callback();
+    return {
+        path: 'tags.json',
+        data: JSON.stringify(tagsMeta)
+    };
   };
 
-}).call(this);
+  var jsons = new Array();
```

Comment:
This line may not be elegant.

Since I don't have much knowledge of JS, if there is any elegant way, please inform me.